alibi/explainers/permutation_importance.py

```diff
@@ -153,18 +153,18 @@ class PermutationImportance(Explainer):
 
     def __init__(self,
                  predictor: Callable[[np.ndarray], np.ndarray],
-                 loss_fns: Optional[
+                 loss_fns: Optional[  # type: ignore[misc, valid-type]
                      Union[
-                         str,
-                         List[str],
+                         Literal[tuple(METRIC_FNS["loss"].keys())],
+                         List[Literal[tuple(METRIC_FNS["loss"].keys())]],
```

Comment:
I don't think `mypy` actually can check this given that the values here are dynamic instead of static (hence the required type-ignore's). On the other hand, [sphinx docs render nicely](https://seldon--798.org.readthedocs.build/projects/alibi/en/798/api/alibi.explainers.html#alibi.explainers.PermutationImportance), docstring on the help message (i.e. `PermutationImportance?`) is properly formatted (!) and we avoid duplicating runtime info and type info. @ascillitoe any thoughts on this?

Reply:
Difficult one... I agree `mypy` will probably not be checking this properly (hence the `type: ignore`'s as you say). That being said the docs look nice and it avoids duplication. 

Is there a reason we can't replace `METRIC_FNS` with a separate Enum for the `loss` and `score` functions?

Reply:
Are you suggesting to use enums to also type the arguments? I've been generally avoiding this as it's not as nice to the user (who would presumably see `score_fns: SCORE_FNS` as a type and have to look up our enum) as opposed to `Literal's

Reply:
Yes that was what I was thinking. But that's a good point it would be more cumbersome when viewing the api docs.

Reply:
Seems like duplication is the only option to satisfy both static analysis and nicely rendered docstrings: https://stackoverflow.com/a/67292548

Reply:
@ascillitoe what do you think about the solution in the link, basically defining the `Literal` type statically (thus duplicating keys) but keeping an `assert` statement to ensure the keys are in sync at runtime? Seems like a reasonable workaround to get both static type-checking working and nice `Literal` docstrings.

Reply:
Sorry missed this. My main concern with the duplication was maintainability (wrt to the keys getting out of sync) rather than extra lines of code, so that solution seems like a really nice idea.

Reply:
ok let's do this @RobertSamoilescu